guess.cpp

```diff
@@ -0,0 +1,63 @@
+#include<iostream>
+#include<fstream>
+/*Vick Ye
+ *1h 51m 41s
+ *10/10
+ *strings, organizing animals and characteristics, nested loops
+ */
+//array representing potential guesses, with inner array representing characteristics
+std::string A[100][100];
```

Comment:
You'll need to learn C-string first.

A C-string is simply a `char` array:
```
char str[101];
```
In order to know how long the string is, a C-string always terminates with a extra char 0 -- often referred as NULL. Thus, we always need 1 extra space for the char array.

To read a C-string:
```
char s[101];
fscanf(In, "%s", s);  // note, s is already an address
```
To compare two C-string:
```
if (strcmp(s1, s2)==0) {
    // they are equal
}
```